### ISSUE TITLE
**ECI to TLE Conversion**

### ISSUE THREAD

**Issue by tpreynolds** [uw-adcs-krish/uw_cubesat_adcs#26]
Secondary software test for this library. Called ''eci2tle_coarse_lib.slx'' under adcs_sim/matlab/tools/.

**Comment by uw-adcs-krish** [comment]
* The function is incorrectly named - it is mapping ECEF to TLE
* Change time_ut1 signal name to datetime_ut1
* Change JD_ut1_days to UTC since that is what SGP4 assumes
* Output UTC days from time_conversions for input to TLE
* **Computing velocity cross product term incorrectly** - should be: ecef2eci * vel_ecef_mps + ppef2veci * pos_ecef_m
  * ppef2veci = ecef2eci * skew(omega_earth) and it operates on r_ecef
  * After changing this, verified to good enough accuracy against my own eci2tle function
  * Since we use ecef2eci and eci2ecef in multiple places for velocity, consider making this a library
* The eci2tle_fcntest.m seems to contain the old version of the code that we were using - delete this, if not being used
* The current simulink unit test eci2tle_test.slx is broken - fix this
* In ECI2OEV matlab function:
  * Delete all the commented code - this makes it very hard to follow the relevant code
  * Use proper indenting
  * Why define your own angle function? Just use atan2
  * Why have parabolic / hyperbolic orbit support? This is unnecessary code space
    * Instead have a couple if conditions on eccentricity at the end to determine validity
  * Add validity flag to the output based on some conditional properties of the elliptic orbit that we will be in, i.e.
    * ecc < 1.0
    * inc > 0.0
    * semi_major_km > 6371
    * etc.
  * rad is a misleading constant definition - should be rad2deg
  * This function should be independent of the inertial frame being used, so don't use r_teme_km / v_teme_kmps inside
  * Delete sqrt_mu and pi2 variables if they aren't being used

**Comment by tpreynolds** [comment]
made all of these changes. Created a unit test for RV2OEV based on ex 2-5 from Vallado and updated the eci2tle_test file. Will include in the next pull request.
